### ISSUE TITLE
Sync should use hashing instead of timestamps

### ISSUE THREAD

**Issue by Bulmanator** [meganz/MEGAsync#609]
The current system that uses timestamps to check if files have been modified or not fails to account for files that may already exist in the sync directory. Thus it will try to upload thousands of identical files because they were copied over beforehand and therefore have newer timestamps.

If hashing was used instead a file would only get synced if the **actual contents** of the file had been changed not the metadata on the filesystem. This would prevent syncing when existing files were already there and when a file may have been touched by an external program but not modified in any capacity.

This would also allow syncing to be setup on existing filesystems that already have all the files, just not the metadata required for MEGAsync to work. This is currently not possible without either re-downloading all of the synced files via MEGAsync or re-syncing the existing files (both which are incredibly slow and redundant)

**Comment by Phileosopher** [comment]
I agree with this. I've had plenty of quirky resync issues because of weird timezone issues from glitches in my VPN. I've also had some horrific experiences trying to run PortableApps.com apps from multiple computers, and this would answer why.

I also have no idea how hard it would be to implement, so I won't severely agree. I'm an edge case user, after all, as evidenced by this comment.

**Comment by rl-mega** [comment]
Hi @Bulmanator and @Phileosopher,
Sorry for the late reply. We have made some great improvements to the sync algorithm since you open this issue.
Please try with the latest version and re-open the ticket if the issue persists. 
Have a good day!
